repository/src/main/java/com/astasko/pdp/repository/api/UserRepository.java

```diff
@@ -0,0 +1,18 @@
+package com.astasko.pdp.repository.api;
+
+import com.astasko.pdp.domain.model.User;
+
+public interface UserRepository extends Repository<User, Long> {
```

Comment:
I think you can use generics for UserRepository
public interface UserRepository<T, I> extends Repository<T, I>

and remove methods from UserRepository interface